tests/components/bryant_evolution/test_climate.py

```diff
@@ -0,0 +1,399 @@
+"""Test the BryantEvolutionClient type."""
+
+import asyncio
+from collections.abc import Callable
+from dataclasses import dataclass
+import logging
+from unittest.mock import patch
+
+from evolutionhttp import BryantEvolutionLocalClient
+import pytest
+
+from homeassistant.components.bryant_evolution.const import CONF_SYSTEM_ZONE, DOMAIN
+from homeassistant.components.climate import (
+    ATTR_FAN_MODE,
+    ATTR_HVAC_ACTION,
+    SERVICE_SET_FAN_MODE,
+    SERVICE_SET_HVAC_MODE,
+    SERVICE_SET_TEMPERATURE,
+    HVACAction,
+    HVACMode,
+)
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.const import ATTR_ENTITY_ID, CONF_FILENAME
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr
+from homeassistant.util.unit_system import US_CUSTOMARY_SYSTEM
+
+from tests.common import MockConfigEntry
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def _wait_for_cond(predicate: Callable[[], bool]) -> None:
+    """Wait for `predicate` to return True."""
+    while True:
+        if predicate():
+            break
+        await asyncio.sleep(0.1)
```

Comment:
I do still have to see where we use this, but we preferably don't want to wait in tests. Except if its completely unavoidable

Reply:
Agree. 

I'm trying to verify that we call self.async_write_ha_state in various update functions in the climate entity. See test_set_fan_mode for an example. I'd love to hear better ways to verify that we see our own updates reflected *before* we execute a read against the device -- is there some HA API I can use?

Reply:
removed